### PR TITLE
CLIO-43950: Modify Kendo scheduler header.

### PR DESCRIPTION
**Pull request by dtomalty**

### CODE REVIEW

File: src/lib/thScheduler/headerBar.template.html [hunk 1]
```diff
@@ -0,0 +1,18 @@
+<div style="width: 100%" />
+  <div id="header-bar">
+    <span>
+      <th-button ng-click="$ctrl.setView('agenda')">Agenda</th-button>
```
Comment (slapheadted) [inline src/lib/thScheduler/headerBar.template.html]: Was there a reason for choosing to replace the existing buttons with our own, instead of appending our custom buttons to their default buttons?

Personally I'd always opt to leave vendor implementation untouched if possible.

File: src/lib/thScheduler/headerBar.template.html [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+      <th-button ng-click="$ctrl.setView('day')">Day</th-button>
```
Comment (slapheadted) [inline src/lib/thScheduler/headerBar.template.html]: We can leverage the `navigate` event for any logic based on current view. [docs link](http://docs.telerik.com/kendo-ui/api/javascript/ui/scheduler#events-navigate) ...should we decide to keep the default buttons.

If we decide to use our own buttons we can leverage the `view` method to switch views. [docs link](http://docs.telerik.com/kendo-ui/api/javascript/ui/scheduler#methods-view)

File: src/lib/thScheduler/scheduler.component.ts [hunk 3]
```diff
@@ -14,10 +14,14 @@ class SchedulerController {
       delegate: this.delegate
     });
   }
+
+  public setView(view) {
```
Comment (slapheadted) [inline src/lib/thScheduler/scheduler.component.ts]: `view` method would probably replace the need for this method [docs link](http://docs.telerik.com/kendo-ui/api/javascript/ui/scheduler#methods-view)

File: src/lib/thScheduler/scheduler.service.ts [hunk 4]
```diff
@@ -12,15 +12,21 @@ declare const kendo: any;
...
       views: [
         "agenda",
         "day",
         "week",
         "month"
       ],
-      dataSource: options["delegate"]["dataSource"]
+      dataSource: options["delegate"]["dataSource"],
+      dataBound: function(e: any) {
+        var buttons = $(document).find("#header-bar").detach();
```
Comment (slapheadted) [inline src/lib/thScheduler/scheduler.service.ts]: Nice find on the `dataBound` callback.

Echoing previous comments, is there a strong reason for replacing the default buttons instead of appending to?

File: src/lib/thScheduler/headerBar.template.html (hunk 2 above)
Comment (slapheadted) [inline src/lib/thScheduler/headerBar.template.html]: As discussed earlier, I now understand that you are essentially de-coupling the Kendo implementation from our thScheduler. My previous comments are no longer applicable :)

File: src/lib/thScheduler/scheduler.component.ts (hunk 3 above)
Comment (slapheadted) [inline src/lib/thScheduler/scheduler.component.ts]: Again, I overlooked the fact that you were de-coupling Kendo from thSchduler 👍

File: src/lib/thScheduler/headerBar.template.html (hunk 1 above)
Comment (dtomalty) [inline src/lib/thScheduler/headerBar.template.html]: As discussed, I think there's an argument for replacing the kendo header controls with th-components. It would minimize duplicate styling, increase maintainability of the header code, and reduce the dependence on Kendo.

File: src/lib/thScheduler/scheduler.service.ts (hunk 4 above)
Comment (dtomalty) [inline src/lib/thScheduler/scheduler.service.ts]: Same as above.

File: src/lib/thScheduler/headerBar.template.html (hunk 1 above)
Comment (slapheadted) [inline src/lib/thScheduler/headerBar.template.html]: Yep 👍 

As we discussed earlier, if the default buttons had significant functionality tied to them - this decision might have been harder to make. But in this case, as we've noticed, it looks like the buttons only need to call the "setView" method; pretty isolated from anything else meaning a cleaner swap-out.

File: src/lib/thScheduler/headerBar.template.html (hunk 2 above)
Comment (slapheadted) [inline src/lib/thScheduler/headerBar.template.html]: ✅

File: src/lib/thScheduler/scheduler.component.ts (hunk 3 above)
Comment (slapheadted) [inline src/lib/thScheduler/scheduler.component.ts]: ✅

File: src/lib/thScheduler/scheduler.service.ts (hunk 4 above)
Comment (slapheadted) [inline src/lib/thScheduler/scheduler.service.ts]: ✅

**Comment by slapheadted** [comment]
Looking good man.
